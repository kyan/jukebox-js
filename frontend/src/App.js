import React from 'react'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import { PersistGate } from 'redux-persist/integration/react'
import jukeboxMiddleware from './containers/jukebox-middleware'
import jukeboxApp from './reducers'
import { Container } from 'semantic-ui-react'
import Dashboard from './containers/dashboard'

const persistConfig = {
  key: 'root',
  storage: storage,
  whitelist: ['settings']
}
const persistedReducer = persistReducer(persistConfig, jukeboxApp)
const store = createStore(persistedReducer, applyMiddleware(jukeboxMiddleware))
const persistor = persistStore(store)

const App = () => (
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <Container fluid>
        <Dashboard />
      </Container>
    </PersistGate>
  </Provider>
)

export default App
