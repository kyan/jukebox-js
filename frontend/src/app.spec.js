import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
jest.mock('react-use-googlelogin')

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
