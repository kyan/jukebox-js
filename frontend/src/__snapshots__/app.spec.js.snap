// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Container
    fluid={true}
  >
    <ContextProvider>
      <ErrorBoundary>
        <_default
          options={Object {}}
        />
        <DragDropContext(Dashboard) />
      </ErrorBoundary>
    </ContextProvider>
  </Container>
</Provider>
`;
